#!/usr/bin/env python
PACKAGE = "CORC"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Window size for the EMG reader
gen.add("calibration_period", int_t, 0, "Number of seconds to calibrate for", 30,  0, 60)

# Window size for the EMG reader
gen.add("EMG_window_size", int_t, 0, "Number of EMG samples to average over", 500,  1, 1000)

# Muscle activation window size
gen.add("muscle_activation_window_size", int_t, 0, "Number of muscle activation samples to average over", 30,  1, 100)

# Stiffness method
stiffness_enum = gen.enum([
                    gen.const("max", int_t, 0, "Max muscle contraction"),
                    gen.const("min", int_t, 1, "Min muscle contraction"),
                    gen.const("high_low", int_t, 2, "High-Low method"),
                    gen.const("contribution", int_t, 3, "Contribution method"),
                    gen.const("just_knee", int_t, 4, "Just knee")],
                    "Stiffness calculation method")

gen.add("stiffness_method", int_t, 0, "Stiffness calculation method", 0, 0, 4, edit_method=stiffness_enum)

muscle_calibration_enum = gen.enum([
                            gen.const("Front_Left_Thigh", int_t, 0, "Front Left Thigh"),
                            gen.const("Back_Left_Thigh", int_t, 1, "Back Left Thigh"),
                            gen.const("Left_Glute", int_t, 2, "Left Glute"),
                            gen.const("Front_Right_Thigh", int_t, 3, "Front Right Thigh"),
                            gen.const("Back_Right_Thigh", int_t, 4, "Bakc Right Thigh"),
                            gen.const("Right_Glute", int_t, 5, "Right Glute")],
                            "Muscle Calibration Selection")

gen.add("cur_muscle", int_t, 0, "Muscle to calibrate", 0, 0, 5, edit_method=muscle_calibration_enum)

exit(gen.generate(PACKAGE, "emgReader", "EMGReader"))
